package/addons-update.sh

```diff
@@ -65,7 +75,30 @@ for f in $(find /etc/kubernetes/addons -name '*.yaml'); do
   sed -i "s|\$HELM_IMAGE_NAMESPACE|$HELM_IMAGE_NAMESPACE|g" ${f}
   sed -i "s|\$DNS_CLUSTER_IP|$DNS_CLUSTER_IP|g" ${f}
   sed -i "s|\$ADDONS_LOG_VERBOSITY_LEVEL|$ADDONS_LOG_VERBOSITY_LEVEL|g" ${f}
-  kubectl --namespace=kube-system replace --force -f ${f}
+  sed -i "s|\$DASHBOARD_IMAGE|$DASHBOARD_IMAGE|g" ${f}
+  sed -i "s|\$KUBEDNS_IMAGE|$KUBEDNS_IMAGE|g" ${f}
+  sed -i "s|\$GRAFANA_IMAGE|$GRAFANA_IMAGE|g" ${f}
+  sed -i "s|\$HEAPSTER_IMAGE|$HEAPSTER_IMAGE|g" ${f}
+  sed -i "s|\$INFLUXDB_IMAGE|$INFLUXDB_IMAGE|g" ${f}
+  sed -i "s|\$TILLER_IMAGE|$TILLER_IMAGE|g" ${f}
+done
+
+addons_images=(
+    "k8s-app=kubernetes-dashboard,$DASHBOARD_IMAGE,/etc/kubernetes/addons/dashobard"
+    "k8s-app=kube-dns,$KUBEDNS_IMAGE,/etc/kubernetes/addons/dns"
+    "k8s-app=grafana,$GRAFANA_IMAGE,/etc/kubernetes/addons/heapster/grafana"
+    "k8s-app=heapster,$HEAPSTER_IMAGE,/etc/kubernetes/addons/heapster/heapster"
+    "k8s-app=influxdb,$INFLUXDB_IMAGE,/etc/kubernetes/addons/heapster/influxdb"
+    "app=helm,$TILLER_IMAGE,/etc/kubernetes/addons/helm"
+   )
+
+# Check Addon version
+for i in "${addons_images[@]}"; do
+  current_version=$(kubectl get deployments -n kube-system -o=jsonpath="{..image}" -l "$(echo $i | cut -d"," -f1)" | cut -d" " -f1 | cut -d":" -f2)
```

Comment:
Do we take care of the case where we fail to fetch the image info by some reason.

Reply:
I added a check if the current version is empty to replace the addon anyway